forest/src/daemon.rs

```diff
@@ -102,7 +103,9 @@ pub(super) async fn start(config: Config) {
     // Start Prometheus server port
     let prometheus_listener = TcpListener::bind(config.metrics_address)
         .await
-        .unwrap_or_else(|_| panic!("could not bind to {}", config.metrics_address));
+        .unwrap_or_else(|_| {
+            cli_error_and_die(&format!("could not bind to {}", config.metrics_address), 1)
```

Comment:
minor suggestion, we could use `AsRef<str>` as an argument to this function to make it a bit more ergonomic to use (e.g. being able to pass just `String`)